### PR TITLE
fuzz: Add libfuzzer compatible bounds fuzzer

### PR DESCRIPTION
**Pull request by silvergasp**

### CODE REVIEW

**Comment by silvergasp** [comment]
See #4088

**Review by rootjalex — COMMENTED** [review]
This is great, thanks for doing this! Lgtm pending green buildbots.

**Review by rootjalex — APPROVED** [review]

File: test/fuzz/bounds.cpp [hunk 1]
```diff
@@ -96,108 +97,98 @@ Expr random_condition(Type T, int depth, bool maybe_scalar) {
     };
     const int op_count = sizeof(make_bin_op) / sizeof(make_bin_op[0]);
```
Comment (steven-johnson) [inline test/fuzz/bounds.cpp:98]: This is unused, it must be removed

**Comment by steven-johnson** [comment]
I took the liberty of harvesting fixes from #7551 and #7548 and applying them here (since our buildbot doesn't test this stuff, I pulled this into Google to test)... unfortunately it's still broken, as running it with default args ends up trying to construct a binary operation with vector elements with mismatched lanes. Gonna debug some more.

File: test/fuzz/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ tests(GROUPS fuzz
       SOURCES
       simplify.cpp
       cse.cpp
+      bounds.cpp
```
Comment (steven-johnson) [inline test/fuzz/CMakeLists.txt:5]: Please sort these alphabetically :-)

**Comment by steven-johnson** [comment]
hrm, this may have found a real bug: it seems we are generating two uint32x6:

```
// Both are uint32x6
a=x3(ramp((uint32)b, (uint32)e, 2)) 
b=x2(ramp((uint32)e, (uint32)4117721957, 3))
```

But if we attempt to cast either to uint1x6 (ie, bool), we fail inside cast at IROperator.cpp:1946 because we don't expect the type to be a combination of Broadcast-and-Ramp. Pondering the right way to handle this.

**Comment by steven-johnson** [comment]
See https://github.com/halide/Halide/issues/7556 for possible fix

**Comment by steven-johnson** [comment]
LGTM. There are still issues to be resolved with the fuzzers but this change seems OK.

**Comment by steven-johnson** [comment]
ready to land?

**Comment by silvergasp** [comment]
All good on my end!

**Pull request merged**